src/syntax_mapping.rs

```diff
@@ -26,14 +26,15 @@ pub enum MappingTarget<'a> {
 #[derive(Debug, Clone, Default)]
 pub struct SyntaxMapping<'a> {
     mappings: Vec<(GlobMatcher, MappingTarget<'a>)>,
+    pub ignored_suffixes: IgnoredSuffixes,
 }
 
 impl<'a> SyntaxMapping<'a> {
     pub fn empty() -> SyntaxMapping<'a> {
         Default::default()
     }
 
-    pub fn builtin() -> SyntaxMapping<'a> {
+    pub fn builtin(ignored_suffixes: Vec<String>) -> SyntaxMapping<'a> {
```

Comment:
Leave this public API unchanged. Instead add an associated function that allows clients to add ignored suffixes. Just like there is public API to add custom syntax mappings.